qml/pages/GalleryUI.qml

```diff
@@ -1,22 +1,100 @@
 import QtQuick 2.0
 import Sailfish.Silica 1.0
+import Nemo.FileManager 1.0
 
 Page {
     id: galleryPage
 
-    property var photoList: ({})
+    property var fileList: ({})
 
     // The effective value will be restricted by ApplicationWindow.allowedOrientations
     allowedOrientations: Orientation.All
 
+    backNavigation: false
+
+    function removeFile(idx) {
+        var path = fileList.get(idx).filePath
+        console.log("Removing", path)
+        FileEngine.deleteFiles( [ path ] )
+        fileList.remove(idx)
+        if (gallery.count === 0) {
+            console.log("Closing empty gallery!")
+            pageStack.pop()
+        }
+    }
+
+    Rectangle {
+        id: boxTop
+
+        z: 99
+        visible: true
+        width: parent.width
+        height: parent.height / 6
```

Comment:
Use item size defines from Theme instead of sizes based on display size. After removing the useless Rectangles used for layout the size would be defined in IconButton (or RoundButton if we want consistent button appearance).

Reply:
Changed for RoundButton, and used Theme.itemSizeMedium so they're a bit bigger.